apps/x509.c

```diff
@@ -1097,3 +1105,93 @@ static int purpose_print(BIO *bio, X509 *cert, X509_PURPOSE *pt)
     }
     return 1;
 }
+
+static int parse_ext_names(char *names, const char **result)
+{
+    char *p, *q;
+    int cnt = 0, idx = 0, len = 0;
+
+    p = q = names;
+    len = strlen(names) + 1;
+
+    while (q - names <= len) {
```

Comment:
How about this to simplify the loop:

```
while ((q = strchr(p, ',')) != NULL) {
    /* found */
```

This also gets rid of the initialisation of `q` above and `len` entirely.

Is it worth considering a check for an empty name.  I.e. `*p == '\0'`  Either return an error or skip the blank name.  Not a request, just a thought.